test/test_nn.py

```diff
@@ -766,6 +748,22 @@ def bw_hook(module, grad_input, grad_output):
         expected_grad = torch.ones(5, 5).mm(module.weight.data) * 2
         self.assertEqual(input.grad.data, expected_grad)
 
+    def test_hook_mutations(self):
+        module = nn.Linear(5, 5)
+        input = torch.randn(5, 5)
+
+        def forward_pre_hook(m, input):
+            return (torch.nn.functional.relu(input[0]),)
+
+        def forward_hook(m, input, output):
+            return -output
+
+        module.register_forward_pre_hook(forward_pre_hook)
+        module.register_forward_hook(forward_hook)
+        output = module(input)
+        expected_res = -torch.nn.functional.linear(torch.nn.functional.relu(input), module.weight, module.bias)
+        self.assertEqual(output, expected_res)
```

Comment:
Does backward get computed correctly now? This is important for QAT.  Previously it would not have mattered as hooks did not modify input or output. Can you add a test for backward correctness?

Reply:
Hooks are no different than any other Python code, so they don't really affect how backward is computed. We don't need tests for that.

Reply:
Yeah we just record the backward on Tensors, so I think it will compute backward correctly as long as the fakequant gradient is correct.